### PR TITLE
fix: fix tx weight calculus when receiving parents in txData

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

**Review by andreabadesso — APPROVED** [review]

File: src/transaction.js [hunk 1]
```diff
@@ -440,7 +440,9 @@ const transaction = {
       throw new MaximumNumberParentsError(`Transaction has ${txData.parents.length} parents and can have at most 2.`);
     }
 
-    txSize += 64 - (32 * (get(txData, 'parents') || 0));
+    // Parents may not be sent in txData
+    const parents = get(txData, 'parents');
```
Comment (msbrogli) [inline src/transaction.js:444]: Why don't we do `const parents = txData.parents || []`?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/transaction.js (hunk 1 above)
Comment (pedroferreira1) [reply src/transaction.js:444]: Using `get` we prevent any error thrown if `txData` is `null` or `undefined`. We are not adding any new dependencies to the project, so I think it's ok to use the `get`.

Comment (msbrogli) [reply src/transaction.js:444]: Ok. But we can still use `|| []` to make sure it is never null, right? Using this we don't have to use the conditional on the line below.

**Review by msbrogli — APPROVED** [review]

File: src/transaction.js (hunk 1 above)
Comment (pedroferreira1) [reply src/transaction.js]: Makes sense. Added it

**Pull request merged**